### PR TITLE
R4R: Feature/0.9.3(Test API)

### PR DESCRIPTION
**Pull request by wade-liwei**

### CODE REVIEW

File: backend/model/block.go [hunk 1]
```diff
@@ -19,3 +21,38 @@ type ValInfo struct {
...
+	BlockHeight         string         `json:"block_height"`
+	Timestamp           time.Time      `json:"timestamp"`
+	BlockHash           string         `json:"block_hash"`
+	Transactions        string         `json:"transactions"`
+	PropopserMoniker    string         `json:"propopser_moniker"`
+	PropoperAddr        string         `json:"propopser_addr"`
+	Rewards             document.Coins `json:"rewards"`
+	LastBlock           int64          `json:"last_block"`
+	LastBlockHash       string         `json:"last_block_hash"`
+	PrecommitValidators string         `json:"precommit_validators"`
```
Comment (zhiqiang-bianjie) [inline backend/model/block.go]: PrecommitValidators --> voteValidatorNum/validatorSetNum

File: backend/model/block.go [hunk 2]
```diff
@@ -19,3 +21,38 @@ type ValInfo struct {
...
+	VotingPower         string         `json:"voting_power"`
```
Comment (zhiqiang-bianjie) [inline backend/model/block.go]: VotingPower --> VotingPowerPercent

File: backend/model/proposal.go [hunk 3]
```diff
@@ -25,6 +26,11 @@ type Proposal struct {
 	Threshold       string         `json:"threshold,omitempty"`
 }
 
+type VotesValidators struct {
```
Comment (zhiqiang-bianjie) [inline backend/model/proposal.go]: Remove unless code

File: backend/rest/controller/block.go [hunk 4]
```diff
@@ -35,14 +38,13 @@ var block = Block{
...
+		block.SetTid(request.TraceId)
+
+		height, err := strconv.ParseInt(Var(request, "height"), 10, 0)
+		if err != nil || height < 1 {
 			panic(types.CodeInValidParam)
-			return nil
 		}
-		block.SetTid(request.TraceId)
-		result := block.Query(height)
+		result := block.QueryBlock(height)
```
Comment (zhiqiang-bianjie) [inline backend/rest/controller/block.go]: method "QueryBlock" --> Query

File: backend/service/service_block.go [hunk 5]
```diff
@@ -16,15 +22,108 @@ func (service *BlockService) GetModule() Module {
...
+	items := []model.BlockValidator{}
+	for k, v := range lcdValidators.Validators {
+		if k >= page*size && k < (page+1)*size {
+			var tmp model.BlockValidator
+			tmp.Consensus = v.PubKey
+			tmp.VotingPower = v.VotingPower
+			tmp.ProposerPriority = v.ProposerPriority
+			var validatorDoc document.Validator
+			var selector = bson.M{"description.moniker": 1, "operator_address": 1}
+			err := queryOne(document.CollectionNmValidator, selector, bson.M{"consensus_pubkey": v.PubKey}, &validatorDoc)
```
Comment (zhiqiang-bianjie) [inline backend/service/service_block.go]: Find all validators at once

File: backend/service/service_block.go [hunk 6]
```diff
@@ -16,15 +22,108 @@ func (service *BlockService) GetModule() Module {
...
 	}
-	return buildBlock(result)
+
+	result.BlockHash = b.BlockMeta.BlockID.Hash
+	result.BlockHeight = b.Block.Header.Height
+
+	heightAsInt64, err := strconv.ParseInt(b.Block.Header.Height, 10, 0)
+
+	if err != nil {
+		logger.Error("block header height is not int type")
```
Comment (zhiqiang-bianjie) [inline backend/service/service_block.go]: panic(types.CodeNotFound)

File: backend/service/service_block.go [hunk 7]
```diff
@@ -16,15 +22,108 @@ func (service *BlockService) GetModule() Module {
...
+
+	if err != nil {
+		logger.Error("query validator collection  err", logger.String("error", err.Error()), service.GetTraceLog())
+		return result
+	}
+	result.PropoperAddr = validatorDoc.OperatorAddress
+	result.PropopserMoniker = validatorDoc.Description.Moniker
+	result.Timestamp = b.BlockMeta.Header.Time
+	result.Transactions = b.BlockMeta.Header.NumTxs
+	if height <= 1 {
```
Comment (zhiqiang-bianjie) [inline backend/service/service_block.go]: Judging at the top of the method(height == 1)

File: backend/service/service_block.go [hunk 8]
```diff
@@ -16,15 +22,108 @@ func (service *BlockService) GetModule() Module {
...
+	err = queryOne(document.CollectionNmValidator, selector, bson.M{"proposer_addr": b.BlockMeta.Header.ProposerAddress}, &validatorDoc)
+
+	if err != nil {
+		logger.Error("query validator collection  err", logger.String("error", err.Error()), service.GetTraceLog())
+		return result
+	}
+	result.PropoperAddr = validatorDoc.OperatorAddress
+	result.PropopserMoniker = validatorDoc.Description.Moniker
+	result.Timestamp = b.BlockMeta.Header.Time
+	result.Transactions = b.BlockMeta.Header.NumTxs
```
Comment (zhiqiang-bianjie) [inline backend/service/service_block.go]: "result.Timestamp = b.BlockMeta.Header.Time
result.Transactions = b.BlockMeta.Header.NumTxs" ,These codes are moved to the front

File: backend/service/service_proposal.go [hunk 9]
```diff
@@ -17,6 +19,32 @@ func (service *ProposalService) GetModule() Module {
 	return Proposal
 }
 
+func (service *ProposalService) QueryProposalsByBlockHeight(height int64) []model.ProposalInfoVo {
```
Comment (zhiqiang-bianjie) [inline backend/service/service_proposal.go]: QueryProposalsByBlockHeight --> QueryProposalsByHeight

File: backend/model/block.go (hunk 1 above)
Comment (wade-liwei) [reply backend/model/block.go]: ok

**Pull request merged**